files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,537 @@ class AaaCfg(object):
+                syslog.syslog(syslog.LOG_INFO, "cmd - {}".format(cmd))
+                try:
+                    subprocess.check_call(cmd, shell=True)
+                except subprocess.CalledProcessError as err:
+                    syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+                    pass
+
+        return True
+
+    def handle_mgmt_intf_creation(self, mgmt_intf_key):
+        syslog.syslog(syslog.LOG_INFO, "Received MGMT Intf creation, key: {}".format(mgmt_intf_key))
+        keys = self.config_db.get_keys(CFG_MGMT_VRF)
+        if keys:
+            syslog.syslog(syslog.LOG_INFO, "MGMT VRF created")
+            self.bind_mgmt_intf()
+
+    def handle_mgmt_vrf_cfg(self, key, data):
+        syslog.syslog(syslog.LOG_INFO, "Received MGMT VRF config, key: {}, data: {}".format(key, data))
+        try:
+            if data:
+                if data['mgmtVrfEnabled'] == 'true':
+                    self.configure_mgmt_vrf()
+                else:
+                    self.unconfigure_mgmt_vrf()
+            else:
+                self.unconfigure_mgmt_vrf()
+        except:
+            syslog.syslog(syslog.LOG_ERR, "{} MGMT VRF Configuration failed.".format(key))
+
+
+
+class MgmtIntfCfg:
+    def __init__(self, cfg_db, mgmt_vrf_cfg):
+        self.app_db = ConfigDBConnector()
+        self.app_db.db_connect('APPL_DB', wait_for_init=False, retry_on=True)
+        self.config_db = cfg_db
+        self.mgmt_vrf_cfg = mgmt_vrf_cfg
+        syslog.syslog(syslog.LOG_INFO, 'AppDB connect success')
+
+        self.mgmt_intf_handle_default_cfg("eth0")
+
+    def mgmt_intf_handle_default_cfg (self, ifname):
+        mgmt_intf = self.config_db.get_entry(CFG_MGMT_PORT_TABLE, ifname)
+        if mgmt_intf == {}:
+            mgmt_intf['mtu'] = "1500"
+            mgmt_intf['speed'] = "1000"
+            mgmt_intf['admin_status'] = MGMT_INTF_ADMIN_STATUS_DEFAULT
+            mgmt_intf['autoneg'] = "true"
+            mgmt_intf['description'] = "Management0"
+            syslog.syslog(syslog.LOG_INFO, "Default config populated for {}".format(ifname))
+            self.config_db.set_entry(CFG_MGMT_PORT_TABLE, ifname, mgmt_intf)
+
+        self.handle_mgmt_port_cfg(ifname, mgmt_intf)
+
+        mgmt_intf = self.config_db.get_entry(CFG_MGMT_INTF_TABLE, ifname)
+        if mgmt_intf == {}:
+            mgmt_intf['ipv4_dhcp_client'] = "true"
+            mgmt_intf['ipv6_dhcp_client'] = "true"
+            self.config_db.set_entry(CFG_MGMT_INTF_TABLE, ifname, mgmt_intf)
+            self.app_db.set_entry(APP_MGMT_INTF_TABLE, ifname, mgmt_intf)
+
+    def mgmt_intf_mtu_set (self, ifname, mtu, appIntf):
+        cmd = 'ifconfig {} mtu {}'.format(ifname, mtu)
+        syslog.syslog(syslog.LOG_INFO, "Configure MTU, cmd - {}".format(cmd))
+        try:
+            subprocess.check_call(cmd, shell=True)
```

Comment:
I see the `command ` call repeated in multiple places, can you define a function for this and just pass the `command`?

Reply:
Defined common methods and using them.